vector/src/main/java/im/vector/app/features/spaces/explore/SpaceDirectoryFragment.kt

```diff
@@ -77,6 +88,32 @@ class SpaceDirectoryFragment @Inject constructor(
         views.toolbar.title = title
     }
 
+    override fun onPrepareOptionsMenu(menu: Menu) = withState(viewModel) { state ->
+        menu.findItem(R.id.spaceAddRoom)?.let {
+            it.isVisible = state.canAddRooms
+        }
+        menu.findItem(R.id.spaceCreateRoom)?.let {
+            it.isVisible = false // Not yet implemented
+        }
+        super.onPrepareOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.spaceAddRoom -> {
+                withState(viewModel) { state ->
+                    addExistingRooms(state.spaceId)
```

Comment:
I would have let `addExistingRooms` to retrive the spaceId from the state, but OK